include/dynd/bool1.hpp

```diff
@@ -40,6 +40,22 @@ class DYND_API bool1 {
     return *this;
   }
 
+  DYND_CUDA_HOST_DEVICE inline bool1 operator!() const {
+    return bool1(m_value == 0);
+  }
+
+  DYND_CUDA_HOST_DEVICE inline bool1 operator~() const {
+    return bool1(m_value == 0);
+  }
+
+  DYND_CUDA_HOST_DEVICE bool1 operator&&(bool1 &rhs) {
+    return bool1(m_value && rhs.m_value);
+  }
+
+  DYND_CUDA_HOST_DEVICE bool1 operator||(bool1 &rhs) {
+    return bool1(m_value || rhs.m_value);
+  }
+
   DYND_CUDA_HOST_DEVICE bool1 &operator+=(bool1 rhs)
```

Comment:
Okay, I've seen that all over the place. I'll remove it here, but a style PR is probably in order for everything else.